walk-in-interview/src/main/webapp/update-job/script.js

```diff
@@ -432,35 +442,35 @@ submitButton.addEventListener('click', (_) => {
     return;
   }
 
-  const jobDetails = getJobDetailsFromUserInput();
-
-  fetch(API['update-job'], {
-    method: 'PATCH',
-    headers: {'Content-Type': 'application/json'},
-    body: JSON.stringify(jobDetails),
-  })
-      .then((response) => {
-        if (response.status == BAD_REQUEST_STATUS_CODE) {
+  getJobDetailsFromUserInput().then((jobDetails) => {
+    fetch(API['update-job'], {
+      method: 'PATCH',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(jobDetails),
+    })
+        .then((response) => {
+          if (response.status == BAD_REQUEST_STATUS_CODE) {
+            setErrorMessage(/* errorMessageElementId= */'error-message',
+                /* msg= */ UPDATE_JOB_STRINGS['storing-error-message'],
+                /* includesDefault= */false);
+            throw new Error(UPDATE_JOB_STRINGS['storing-error-message']);
+          }
+
+          /* reset the error (there might have been an error msg earlier) */
           setErrorMessage(/* errorMessageElementId= */'error-message',
-              /* msg= */ UPDATE_JOB_STRINGS['storing-error-message'],
-              /* includesDefault= */false);
-          throw new Error(UPDATE_JOB_STRINGS['storing-error-message']);
-        }
-
-        /** reset the error (there might have been an error msg from earlier) */
-        setErrorMessage(/* errorMessageElementId= */'error-message',
-            /* msg= */ '', /* includesDefault= */false);
-        window.location.href= HOMEPAGE_PATH;
-      })
-      .catch((error) => {
-        // Not the server response error already caught and thrown
-        if (error.message != UPDATE_JOB_STRINGS['storing-error-message']) {
-          setErrorMessage(/* errorMessageElementId= */'error-message',
-              /* msg= */ UPDATE_JOB_STRINGS['error-message'],
-              /* includesDefault= */false);
-          console.log('error', error);
-        }
-      });
+              /* msg= */ '', /* includesDefault= */false);
+          window.location.href= HOMEPAGE_PATH;
```

Comment:
if you change the location after setting the error message, i'm pretty sure the error message will not be shown since browser will redirect immediately.